"Class versions" check failed in CMSSW_15_0_X_2025-01-23-1100

The check was marked as failed in CMSSW_15_0_X_2025-01-23-1100, but not in -2300 (but it's been failing for a long time). Logs: [1100](https://cmssdt.cern.ch/SDT/jenkins-artifacts//class_versions/CMSSW_15_0_X_2025-01-23-1100/el8_amd64_gcc12/class_versions/CondFormats/L1TObjects/edmDumpClassVersion.html), [2300](https://cmssdt.cern.ch/SDT/jenkins-artifacts/class_versions/CMSSW_15_0_X_2025-01-23-2300/el8_amd64_gcc12/class_versions/CondFormats/L1TObjects/edmDumpClassVersion.html)

I think the class dictionary isn't generated properly: `    std::pair > dummy13;`

@makortel FYI

When I run the `edmDumpClassVersion` command in a freshly made `CMSSW_15_0_X_2025-01-23-1100` which does not contain any checked out code, the program fails, but does not seg fault.

I get the same error messages with the same dump, but I also get additional error messages

```
Error in <TClass::LoadClassInfo>: no interpreter information for class L1TMuonBarrelParams is available even though it has a TClass initialization routine.
Error in <TClass::LoadClassInfo>: no interpreter information for class L1TMuonBarrelParams is available even though it has a TClass initialization routine.
Error in <TInterpreter::AutoParse>: Error parsing payload code for class L1TMuonBarrelKalmanParams with content:

[cut full classes.h content]


In module 'std' imported from input_line_1:1:
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02873/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/bin/../lib/gcc/x86_64-redhat-linux-gnu/12.3.1/../../../../include/c++/12.3.1/bits/stl_pair.h:746:14: error: call to constructor of '__pair_type' (aka 'pair<std::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::basic_string<char, std::char_traits<char>, std::allocator<char> > >') is ambiguous
      return __pair_type(std::forward<_T1>(__x), std::forward<_T2>(__y));
             ^           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/cvmfs/cms-ib.cern.ch/sw/x86_64/week1/el8_amd64_gcc12/cms/cmssw-patch/CMSSW_15_0_X_2025-01-23-1100/src/CondFormats/L1TObjects/interface/L1TriggerKey.h:48:33: note: in instantiation of function template specialization 'std::make_pair<const std::basic_string<char, std::char_traits<char>, std::allocator<char> > &, const std::basic_string<char, std::char_traits<char>, std::allocator<char> > &>' requested here
      m_recordToKey.insert(std::make_pair(itr->first, itr->second.empty() ? kNullKey : itr->second));
                                ^
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02873/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/bin/../lib/gcc/x86_64-redhat-linux-gnu/12.3.1/../../../../include/c++/12.3.1/bits/stl_pair.h:269:7: note: candidate constructor
      pair(const _T1& __x, const _T2& __y)
      ^
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02873/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/bin/../lib/gcc/x86_64-redhat-linux-gnu/12.3.1/../../../../include/c++/12.3.1/bits/stl_pair.h:269:7: note: candidate constructor
In file included from CondFormatsL1TObjects_xr dictionary payload:87:
/cvmfs/cms-ib.cern.ch/sw/x86_64/week1/el8_amd64_gcc12/cms/cmssw-patch/CMSSW_15_0_X_2025-01-23-1100/src/CondFormats/L1TObjects/interface/L1TriggerKeyExt.h:48:28: error: no matching function for call to 'make_pair'
      m_recordToKey.insert(std::make_pair(itr->first, itr->second.empty() ? kNullKey : itr->second));
                           ^~~~~~~~~~~~~~
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02873/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/bin/../lib/gcc/x86_64-redhat-linux-gnu/12.3.1/../../../../include/c++/12.3.1/bits/stl_pair.h:741:5: note: candidate template ignored: substitution failure [with _T1 = const std::basic_string<char, std::char_traits<char>, std::allocator<char> > &, _T2 = const std::basic_string<char, std::char_traits<char>, std::allocator<char> > &]
    make_pair(_T1&& __x, _T2&& __y)
    ^
In file included from CondFormatsL1TObjects_xr dictionary payload:113:
/cvmfs/cms-ib.cern.ch/sw/x86_64/week1/el8_amd64_gcc12/cms/cmssw-patch/CMSSW_15_0_X_2025-01-23-1100/src/CondFormats/L1TObjects/interface/L1TMuonGlobalParams.h:376:24: error: redefinition of 'operator<<'
  friend std::ostream& operator<<(std::ostream& o, const L1TMuonGlobalParams& p) {
                       ^
/cvmfs/cms-ib.cern.ch/sw/x86_64/week1/el8_amd64_gcc12/cms/cmssw-patch/CMSSW_15_0_X_2025-01-23-1100/src/CondFormats/L1TObjects/interface/L1TMuonGlobalParams.h: note: previous definition is here
In module 'std' imported from input_line_1:1:
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02873/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/bin/../lib/gcc/x86_64-redhat-linux-gnu/12.3.1/../../../../include/c++/12.3.1/ext/aligned_buffer.h:77:40: error: 'this' argument to member function '_M_addr' has type 'const __gnu_cxx::__aligned_membuf<std::pair<const int, std::basic_string<char, std::char_traits<char>, std::allocator<char> > > >', but function is not marked const
      { return static_cast<const _Tp*>(_M_addr()); }
                                       ^~~~~~~
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02873/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/bin/../lib/gcc/x86_64-redhat-linux-gnu/12.3.1/../../../../include/c++/12.3.1/bits/stl_tree.h:239:27: note: in instantiation of member function '__gnu_cxx::__aligned_membuf<std::pair<const int, std::basic_string<char, std::char_traits<char>, std::allocator<char> > > >::_M_ptr' requested here
      { return _M_storage._M_ptr(); }
                          ^
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02873/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/bin/../lib/gcc/x86_64-redhat-linux-gnu/12.3.1/../../../../include/c++/12.3.1/bits/stl_tree.h:363:50: note: in instantiation of member function 'std::_Rb_tree_node<std::pair<const int, std::basic_string<char, std::char_traits<char>, std::allocator<char> > > >::_M_valptr' requested here
      { return static_cast<_Link_type>(_M_node)->_M_valptr(); }
                                                 ^
/cvmfs/cms-ib.cern.ch/sw/x86_64/week1/el8_amd64_gcc12/cms/cmssw-patch/CMSSW_15_0_X_2025-01-23-1100/src/CondFormats/L1TObjects/interface/L1TriggerKeyExt.h:73:47: note: in instantiation of member function 'std::_Rb_tree_const_iterator<std::pair<const int, std::basic_string<char, std::char_traits<char>, std::allocator<char> > > >::operator->' requested here
    return key == m_subsystemKeys.end() || key->second == kNullKey ? kEmptyKey : key->second;
                                              ^
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02873/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/bin/../lib/gcc/x86_64-redhat-linux-gnu/12.3.1/../../../../include/c++/12.3.1/ext/aligned_buffer.h:64:7: note: '_M_addr' declared here
      _M_addr() noexcept
      ^
Error in <TInterpreter::AutoParse>: Error parsing payload code for class L1GtAlgorithm with content:

[cut full classes.h content]


Error in <TClass::LoadClassInfo>: no interpreter information for class map<string,L1GtAlgorithm> is available even though it has a TClass initialization routine.
Ignoring class std::map<std::string, L1GtAlgorithm>::value_type as could not get its version and checksum, because: failed to load dictionary for class 'std::map<std::string, L1GtAlgorithm>::value_type'
```

--- a/CondFormats/L1TObjects/interface/L1TMuonBarrelParams.h
+++ b/CondFormats/L1TObjects/interface/L1TMuonBarrelParams.h
@@ -7,12 +7,13 @@
 /// \author: Giannis Flouris
 ///
 
-#ifndef L1TBMTFParams_h
-#define L1TBMTFParams_h
+#ifndef CondFormats_L1TObjects_L1TMuonBarrelParams_h
+#define CondFormats_L1TObjects_L1TMuonBarrelParams_h
 
 #include <memory>
 #include <iostream>
 #include <vector>
+#include <map>
 
 #include "CondFormats/Serialization/interface/Serializable.h"
 #include "CondFormats/L1TObjects/interface/LUT.h"

--- a/CondFormats/L1TObjects/interface/L1TriggerKey.h
+++ b/CondFormats/L1TObjects/interface/L1TriggerKey.h
@@ -40,12 +40,12 @@ public:
 
   /* Adds new record and type mapping to payload. If such exists, nothing happens */
   void add(const std::string& record, const std::string& type, const std::string& key) {
-    m_recordToKey.insert(std::make_pair(record + "@" + type, key.empty() ? kNullKey : key));
+    m_recordToKey.emplace(record + "@" + type, key.empty() ? kNullKey : key);
   }
 
   void add(const RecordToKey& map) {
     for (RecordToKey::const_iterator itr = map.begin(); itr != map.end(); ++itr) {
-      m_recordToKey.insert(std::make_pair(itr->first, itr->second.empty() ? kNullKey : itr->second));
+      m_recordToKey.emplace(itr->first, itr->second.empty() ? kNullKey : itr->second);
     }
   }
 

--- a/CondFormats/L1TObjects/interface/L1TriggerKeyExt.h
+++ b/CondFormats/L1TObjects/interface/L1TriggerKeyExt.h
@@ -40,12 +40,12 @@ public:
 
   /* Adds new record and type mapping to payload. If such exists, nothing happens */
   void add(const std::string& record, const std::string& type, const std::string& key) {
-    m_recordToKey.insert(std::make_pair(record + "@" + type, key.empty() ? kNullKey : key));
+    m_recordToKey.emplace(record + "@" + type, key.empty() ? kNullKey : key);
   }
 
   void add(const RecordToKey& map) {
     for (RecordToKey::const_iterator itr = map.begin(); itr != map.end(); ++itr) {
-      m_recordToKey.insert(std::make_pair(itr->first, itr->second.empty() ? kNullKey : itr->second));
+      m_recordToKey.emplace(itr->first, itr->second.empty() ? kNullKey : itr->second);
     }
   }
 

